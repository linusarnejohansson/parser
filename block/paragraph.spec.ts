import { Error, IO } from "@cogneco/mend"
import { Source } from "../Source"
import { parse } from "."

describe("block.paragraph", () => {
	it("simple", () => {
		const result = parse(new Source(IO.StringReader.create("This is a single paragraph."), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
})

describe("block.definitionList", () => {
	it("simple", () => {
		const result = parse(new Source(IO.StringReader.create("Term 1\n: Description 1\n: Description A\nTerm 2\n: Description 2\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
})
