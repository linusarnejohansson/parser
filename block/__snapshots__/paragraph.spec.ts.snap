// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block.definitionList simple 1`] = `
Array [
  Object {
    "class": "Block.DefinitionList",
    "content": Array [
      Object {
        "class": "Block.DefinitionTerm",
        "content": Array [
          Object {
            "class": "Inline.Text",
            "value": "Term 1
",
          },
        ],
        "data": Array [
          Object {
            "class": "Block.DefinitionData",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "Description 1
",
              },
            ],
          },
          Object {
            "class": "Block.DefinitionData",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "Description A
",
              },
            ],
          },
        ],
      },
      Object {
        "class": "Block.DefinitionTerm",
        "content": Array [
          Object {
            "class": "Inline.Text",
            "value": "Term 2
",
          },
        ],
        "data": Array [
          Object {
            "class": "Block.DefinitionData",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "Description 2
",
              },
            ],
          },
        ],
      },
    ],
  },
]
`;

exports[`block.paragraph simple 1`] = `
Array [
  Object {
    "class": "Block.Paragraph",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "This is a single paragraph.",
      },
    ],
  },
]
`;
